providers/plunk/src/lib/plunk.provider.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  IEmailOptions,
+  IEmailProvider,
+  ICheckIntegrationResponse,
+  IEmailEventBody,
+  CheckIntegrationResponseEnum,
+} from '@novu/stateless';
+
+import Plunk from '@plunk/node';
+import { IPlunkResponse } from './plunk.interface';
+
+export class PlunkEmailProvider implements IEmailProvider {
+  id = 'plunk';
+  channelType = ChannelTypeEnum.EMAIL as ChannelTypeEnum.EMAIL;
+
+  private plunk: Plunk;
+
+  constructor(
+    private config: {
+      apiKey: string;
+    }
+  ) {
+    this.plunk = new Plunk(this.config.apiKey);
+  }
+  getMessageId?: (body: any) => string[];
+  parseEventBody?: (body: any, identifier: string) => IEmailEventBody;
+  async checkIntegration(
+    options: IEmailOptions
+  ): Promise<ICheckIntegrationResponse> {
+    try {
+      const response: IPlunkResponse = await this.plunk.emails.send({
+        to: options.to,
+        subject: options.subject,
+        body: options.html || options.text,
+      });
+
+      return {
+        success: response.success,
+        message: 'Integrated successfully!',
+        code: CheckIntegrationResponseEnum.SUCCESS,
+      };
+    } catch (error) {
+      return {
+        success: false,
+        message: error?.message,
+        code: CheckIntegrationResponseEnum.FAILED,
+      };
+    }
+  }
+
+  async sendMessage(
+    options: IEmailOptions
+  ): Promise<ISendMessageSuccessResponse> {
+    const response: IPlunkResponse = await this.plunk.emails.send({
+      from: options.from,
+      to: options.to,
+      subject: options.subject,
+      body: options.html || options.text,
+    });
+
+    return {
+      id: response.contact,
+      date: new Date().toISOString(),
```

Comment:
according to updated provider docs I see there is `response.timestamp`. 
We use `new Date().toISOString()` only as fallback when provider does not return any timestamp.

so I suggest to use  `date: response.timestamp`

Reply:
Yeah. Got that. I'm making the update already

Reply:
@michaldziuba03 I have updated the response to return the timestamp from the provider